### PR TITLE
Onboarding Improvements: Introduce Feature Flag

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent #14989

This PR introduces a local feature flag for the `Onboarding Improvements: Existing Users` feature.

To test:
- Launch the app and go to `My Site` -> `App Settings` -> `Test feature configuration`.
- Make sure the `OnboardingImprovementsFeatureConfig` is available.
- Enable the `OnboardingImprovementsFeatureConfig` and click on the `RESTART THE APP` button at the bottom.
- Make sure the app is working as expected.
- Disable the `OnboardingImprovementsFeatureConfig` and click on the `RESTART THE APP` button at the bottom.
- Make sure the app is working as expected.

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ashiagr — APPROVED** [review]
All looks good to me 👍. 

Let's get started :).

File: WordPress/src/main/java/org/wordpress/android/util/config/OnboardingImprovementsFeatureConfig.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import org.wordpress.android.annotation.FeatureInDevelopment
+import javax.inject.Inject
+
+/**
+ * Configuration of the 'Onboarding Improvement: Existing Users' feature.
+ */
+@FeatureInDevelopment
+class OnboardingImprovementsFeatureConfig @Inject constructor(
+    appConfig: AppConfig
+) : FeatureConfig(appConfig, BuildConfig.GLOBAL_STYLE_SUPPORT)
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/util/config/OnboardingImprovementsFeatureConfig.kt:13]: 👋 @ParaskP7,
 
Sorry for this late comment 🤦‍♀️, I just noticed we need to replace `GLOBAL_STYLE_SUPPORT` with `ONBOARDING_IMPROVEMENTS`. 

Let's replace it as part of the next PR that we submit where we actually use the feature flag.

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/util/config/OnboardingImprovementsFeatureConfig.kt:13]: 👋 @ashiagr !

OMG, such a silly copy-paste mistake, you are totally right! 🤦 

I can also create a quick fix right now?

Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/util/config/OnboardingImprovementsFeatureConfig.kt:13]: okay, I'll approve it right away 👍 so we're not blocked.

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/util/config/OnboardingImprovementsFeatureConfig.kt:13]: 👋 @ashiagr !

I went and created a quick fix after all, no need for us to wait on it: #15003

**Pull request merged**